README.md

```diff
@@ -1,109 +1,134 @@
-
-Note that bebone is only a **set** of useful libraries/modules, some modules can be used separately.
-
-## Testing
-The testing process is similar to the **Building**. After a successful build, all you have to do is run CTest from ```build\tests\unit``` directory.
-
-## Dependencies
-For convenience, we try to reduce the use of third-party libraries, as it complicates code management and readability.
-At the moment, bebone is only dependent on these libraries:
-  - [Glad](https://github.com/Maksasj/glad/tree/all-extensions) *(opengl function header)*
-  - [Vulkan-Headers](https://github.com/Maksasj/Vulkan-Headers/tree/main) *(vulkan function header)*
-  - [GLFW](https://github.com/Maksasj/glfw/tree/master) *(window management library)*
+![GitHub Repo stars](https://img.shields.io/github/stars/Maksasj/bebone)
+![GitHub Actions Workflow Status](https://img.shields.io/github/actions/workflow/status/Maksasj/bebone/clang_build_win.yml)
+![GitHub License](https://img.shields.io/github/license/Maksasj/bebone)
+![Discord](https://img.shields.io/discord/1156584129190711326?logo=discord)
+
+### Links
+1. Source code available at [github.com/Maksasj/bebone](https://github.com/Maksasj/bebone)
+2. Community Discord [discord.gg/bebone](https://discord.gg/v4mcTmuDTb)
+3. **Bebone** documentation available at [github.com/Maksasj/bebone](github.com/Maksasj/bebone/docs/DOCUMENTATION.md)
+
+## Architecture ⚙️
+
+**Bebone** provides multi layer APIs where each level build on top of previous:
+
+- **[Abstraction]()** basic abstraction layer, abstracts most basic things, such as OpenGL or Vulkan
+- **[System]()** advance abstraction layer, implements generalized systems on top of **abstraction layer**, such as **Renderer**
+- **[App]()** - complete application layer, combines multiple systems into self-contained application 
+
+
+
+## Features ✨
+
+For now **bebone** have severall modules:
+
+* [Core module](). Core **bebone** module implements most important abstractions which are widely used across all other modules. Implements **Memory, Debug, Types, Event, Input** abstractions
+* [Graphics module (GFX module)](). **Abstraction layer** module, implements basic abstractions on top of **Graphical APIs** such as **OpenGL** and **Vulkan**. **GFX module**
+ depends on **Core**, **Assets** modules
+* [Assets module](). Module designed for loading and managing different **assets** such as **images**, **models**, **materals**. **Assets module** depends on **Core module** only 
+* [Renderer module]() **System layer** module, provides high-level generalized graphical functionality. It abstracts all low-level **OpenGL** or **Vulkan** under one convenient interface. This module is build on top of **GFX module**
+* [Sound module]() *Todo*
+
+### Planned modules
+* [Physics module]()
+* [Entity component system module (ECS module)]()
+* [App module]()
+
+## Build 🛠
+For now there is only one build method
+
+### Build manually

```

Comment:
I think, this is unnecessary text. We can omit this until the new build opportunities are available